app/main.py

```diff
@@ -1,2 +1,118 @@
 class Dictionary:
-    pass
+    LOAD_FACTOR = 2 / 3
+    RESIZE = 2
+    DEFAULT_CAPACITY = 8
+
+    def __init__(self):
+        self.initial_capacity = 8
+        self.storage = [[] for _ in range(self.DEFAULT_CAPACITY)]
+        self.length = 0
+
+    def resize(self):
+        saved_storage = self.storage[:]
+        self.length = 0
+        self.initial_capacity *= self.RESIZE
+        self.storage = [[] for _ in range(self.initial_capacity)]
+
+        for item in saved_storage:
+            if len(item) != 0:
+                self.__setitem__(item[0], item[1])
+
+    def __setitem__(self, key, value):
+        if self.length == int(self.initial_capacity * self.LOAD_FACTOR):
+            self.resize()
+
+        hashed_value = hash(key)
+        index = hashed_value % self.initial_capacity
+
+        if len(self.storage[index]) == 0:
+            self.storage[index] = [key, value, hashed_value]
+            self.length += 1
+
+        elif hashed_value == self.storage[index][2] and \
+                key == self.storage[index][0]:
+            self.storage[index][1] = value
+
+        else:
+            index = (index + 1) % self.initial_capacity
+
+            while True:
+                if not self.storage[index]:
+                    self.storage[index] = [key, value, hashed_value]
+                    self.length += 1
+                    return
+                elif hashed_value == self.storage[index][2] and \
+                        key == self.storage[index][0]:
+                    self.storage[index][1] = value
+                    return
+                index = (index + 1) % self.initial_capacity
+
+    def __getitem__(self, key):
+        hashed_value = hash(key)
+        index = hashed_value % self.initial_capacity
+
+        for _ in range(self.initial_capacity):
```

Comment:
This loop takes O(n) time for every search of the missing key. You should optimize it :(

Reply:
I understand that you can't raise KeyError when you meet the first empty value because of deletion.
Probably, you should change deletion to fix it

Reply:
Yes, we have O(n) only when we have many collisions